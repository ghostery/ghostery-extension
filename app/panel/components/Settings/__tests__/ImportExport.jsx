/**
 * Import Export Settings Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2020 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import ImportExport from '../ImportExport';

jest.mock('../../../../../src/classes/Globals', () => ({
	BROWSER_INFO: {
		name: 'firefox',
	}
}));

describe('app/panel/Settings/ImportExport.jsx', () => {
	describe('Snapshot tests with react-test-renderer', () => {
		test('ImportExport is rendered correctly with baseline props', () => {
			const settingsData = {
				pageUrl: '',
				exportResultText: '',
				importResultText: '',
				actionSuccess: false,
			};
			const actions = {
				exportSettings: () => {},
				importSettingsDialog: () => {},
				importSettingsNative: () => {},
			};

			const component = renderer.create(
				<ImportExport
					settingsData={settingsData}
					actions={actions}
				/>
			).toJSON();
			expect(component).toMatchSnapshot();
		});

		test('ImportExport is rendered correctly with happy-path props', () => {
			const settingsData = {
				pageUrl: 'https://example.com',
				exportResultText: 'Your settings have been successfully exported to your downloads folder on August 8, 2020 6:08 PM',
				importResultText: 'Your settings have been successfully imported on September 12, 2020 6:08 PM',
				actionSuccess: true,
			};
			const actions = {
				exportSettings: () => {},
				importSettingsDialog: () => {},
				importSettingsNative: () => {},
			};

			const component = renderer.create(
				<ImportExport
					settingsData={settingsData}
					actions={actions}
				/>
			).toJSON();
			expect(component).toMatchSnapshot();
		});

		test('ImportExport is rendered correctly with unhappy-path props', () => {
			const settingsData = {
				pageUrl: 'chrome://extensions',
				exportResultText: 'Ghostery cannot export settings when the current page is one of the reserved browser pages. Please navigate to a different page and try again.',
				importResultText: 'That is the incorrect file type. Please choose a .ghost file and try again.',
				actionSuccess: false,
			};
			const actions = {
				exportSettings: () => {},
				importSettingsDialog: () => {},
				importSettingsNative: () => {},
			};

			const component = renderer.create(
				<ImportExport
					settingsData={settingsData}
					actions={actions}
				/>
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('Shallow snapshot tests rendered with Enzyme', () => {
		test('ImportExport functions correctly', () => {
			const settingsData = {
				pageUrl: '',
				exportResultText: '',
				importResultText: '',
				actionSuccess: false,
			};
			const actions = {
				exportSettings: jest.fn(),
				importSettingsDialog: jest.fn(),
				importSettingsNative: jest.fn(),
			};
			const component = mount(
				<ImportExport
					settingsData={settingsData}
					actions={actions}
				/>
			);

			expect(actions.exportSettings.mock.calls.length).toBe(0);
			expect(actions.importSettingsDialog.mock.calls.length).toBe(0);
			expect(actions.importSettingsNative.mock.calls.length).toBe(0);
			expect(component.find('.export-result').text()).toBe('');
			expect(component.find('.import-result').text()).toBe('');

			component.find('.export').simulate('click');
			component.setProps({
				settingsData: {
					pageUrl: '',
					exportResultText: 'export-result-text',
					importResultText: '',
					actionSuccess: true,
				}
			});
			expect(actions.exportSettings.mock.calls.length).toBe(1);
			expect(component.find('.export-result').text()).toBe('export-result-text');

			component.find('.import').simulate('click');
			component.setProps({
				settingsData: {
					pageUrl: '',
					exportResultText: '',
					importResultText: 'import-result-text',
					actionSuccess: true,
				}
			});
			expect(actions.importSettingsDialog.mock.calls.length).toBe(1);
			expect(component.find('.import-result').text()).toBe('import-result-text');

			component.find('#select-file').simulate('change');
			expect(actions.importSettingsNative.mock.calls.length).toBe(1);
		});
	});
});
