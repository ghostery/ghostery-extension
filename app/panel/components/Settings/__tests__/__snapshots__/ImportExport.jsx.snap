// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/panel/Settings/ImportExport.jsx Snapshot tests with react-test-renderer ImportExport is rendered correctly with baseline props 1`] = `
<div>
  <p
    className="s-blue-header export"
    onClick={[Function]}
  >
    settings_export_header
  </p>
  <p
    className="s-regular "
  >
    settings_export_text
  </p>
  <p
    className="s-regular export-result"
    style={
      Object {
        "color": "red",
      }
    }
  >
    
  </p>
  <div
    className="s-vgap-20"
  />
  <p
    className="s-blue-header import"
    onClick={[Function]}
  >
    settings_import_header
  </p>
  <p
    className="s-regular "
  >
    settings_import_text
  </p>
  <p
    className="s-regular "
  >
    settings_import_warning
  </p>
  <p
    className="s-regular import-result"
    style={
      Object {
        "color": "red",
      }
    }
  >
    
  </p>
  <input
    id="select-file"
    name="select-file"
    onChange={[Function]}
    type="file"
  />
</div>
`;

exports[`app/panel/Settings/ImportExport.jsx Snapshot tests with react-test-renderer ImportExport is rendered correctly with happy-path props 1`] = `
<div>
  <p
    className="s-blue-header export"
    onClick={[Function]}
  >
    settings_export_header
  </p>
  <p
    className="s-regular s-hide"
  >
    settings_export_text
  </p>
  <p
    className="s-regular export-result"
    style={
      Object {
        "color": "#2092BF",
      }
    }
  >
    Your settings have been successfully exported to your downloads folder on August 8, 2020 6:08 PM
  </p>
  <div
    className="s-vgap-20"
  />
  <p
    className="s-blue-header import"
    onClick={[Function]}
  >
    settings_import_header
  </p>
  <p
    className="s-regular s-hide"
  >
    settings_import_text
  </p>
  <p
    className="s-regular s-hide"
  >
    settings_import_warning
  </p>
  <p
    className="s-regular import-result"
    style={
      Object {
        "color": "#2092BF",
      }
    }
  >
    Your settings have been successfully imported on September 12, 2020 6:08 PM
  </p>
  <input
    id="select-file"
    name="select-file"
    onChange={[Function]}
    type="file"
  />
</div>
`;

exports[`app/panel/Settings/ImportExport.jsx Snapshot tests with react-test-renderer ImportExport is rendered correctly with unhappy-path props 1`] = `
<div>
  <p
    className="s-blue-header export"
    onClick={[Function]}
  >
    settings_export_header
  </p>
  <p
    className="s-regular s-hide"
  >
    settings_export_text
  </p>
  <p
    className="s-regular export-result"
    style={
      Object {
        "color": "red",
      }
    }
  >
    Ghostery cannot export settings when the current page is one of the reserved browser pages. Please navigate to a different page and try again.
  </p>
  <div
    className="s-vgap-20"
  />
  <p
    className="s-blue-header import"
    onClick={[Function]}
  >
    settings_import_header
  </p>
  <p
    className="s-regular s-hide"
  >
    settings_import_text
  </p>
  <p
    className="s-regular s-hide"
  >
    settings_import_warning
  </p>
  <p
    className="s-regular import-result"
    style={
      Object {
        "color": "red",
      }
    }
  >
    That is the incorrect file type. Please choose a .ghost file and try again.
  </p>
  <input
    id="select-file"
    name="select-file"
    onChange={[Function]}
    type="file"
  />
</div>
`;
